From 3f5e56bf9b814064104911edb86b6006986d1a00 Mon Sep 17 00:00:00 2001
From: Marcin Niestroj <m.niestroj@emb.dev>
Date: Thu, 13 Jun 2024 17:15:06 +0200
Subject: [PATCH] scripts: pytest: execute 'zephyr.exe' in application build
 directory cwd

Execute 'zephyr.exe' in application build directory as 'current working
directory' (cwd). This makes sure that native_sim specific drivers (like
flash simulator with file backend in 'flash.bin') are using unique context
for external resources with relative paths.

This fixes executing native_sim tests in twister with flash simulator.
Previously a shared 'flash.bin' was used for all executed 'zephyr.exe'
processes in twister. After this patch a unique 'flash.bin' file is used
for each tested sample, since those 'flash.bin' is placed in application
build directory instead of twister root directory.

Signed-off-by: Marcin Niestroj <m.niestroj@emb.dev>
---
 .../src/twister_harness/device/binary_adapter.py                 | 1 +
 1 file changed, 1 insertion(+)

diff --git a/scripts/pylib/pytest-twister-harness/src/twister_harness/device/binary_adapter.py b/scripts/pylib/pytest-twister-harness/src/twister_harness/device/binary_adapter.py
index 228981be0f1..c48d548f9ca 100755
--- a/scripts/pylib/pytest-twister-harness/src/twister_harness/device/binary_adapter.py
+++ b/scripts/pylib/pytest-twister-harness/src/twister_harness/device/binary_adapter.py
@@ -28,6 +28,7 @@ class BinaryAdapterBase(DeviceAdapter, abc.ABC):
             'stderr': subprocess.STDOUT,
             'stdin': subprocess.PIPE,
             'env': self.env,
+            'cwd': device_config.build_dir,
         }
 
     @abc.abstractmethod
-- 
2.45.2

